src/edu/mills/cs180a/RomanNumeral.java

```diff
@@ -14,154 +14,175 @@
+                // has a higher value than the first. Subtract current twice: once because it is the
+                // start of a subtractive form and once because it was added in a previous
+                // iteration.
+                numericValue += LETTERS_TO_VALUES.get(next) - (2 * LETTERS_TO_VALUES.get(current));
+            }
+        }
+        return numericValue;
+    }// end convertFromString
+
+
     /**
-     * Returns the Roman Numeral representation of the given number.
+     * Returns the Roman Numeral representation of any given number by first determining which keys
+     * in {@link #VALUES_TO_LETTERS} the number falls between and setting the lesser key as a
+     * {@link #lowerBound}. The number, {@link #lowerBound} and a StringBuilder are passed to a
+     * helper method, which will append the appropriate Roman symbol to the StringBuilder and return
+     * the number, of which the value has decreased based on the numeric value of the appended
+     * symbol.
      *
      * @param n the number to convert
      * @return the Roman Numeral representation
-     * @throws IllegalArgumentException if the number cannot be represented
-     * by a Roman Numeral in the given range
+     * @throws IllegalArgumentException if the argument is out of bounds
      */
     @VisibleForTesting
-    protected static String convertToText(int n) 
-    {
-    	//GOAL: convert number into strings(Roman Numerals)
-    	//Refer to note in convertToInteger
-    	
-    	//acquire input, compare it to the values in the map
-    	//return the key
-    	if(n==MIN_VALUE)
-    	{
-    		LETTERS_TO_VALUES.containsValue(n);
-    	}
-    	
-    	
- 	 	return null;
-    }
-}//end RomanNumeral class
+    protected static String convertFromInt(int n) {
+        // lowerBound should be set to the Roman Numeral that closely matches the outermost place
+        // value of any given number. Ex n = 56, lowerBound = L; n = 383, lowerBound = C
+        String lowerBound = "";
+        StringBuilder romanNumeral = new StringBuilder();
+
+        if (n > MAX_VALUE || n < MIN_VALUE) {
+            throw new IllegalArgumentException(n + "is out of bounds");
+        }
+
+        // once the value of n reaches 0, the Roman Numeral string is complete
+        while (n > 0) {
+            for (int i = 0; i < NUMERICS.length; i++) {
+                // if the numeric value of the integer is 1000 or greater, which will cause the loop
+                // to reach the end of NUMERICS, set lowerBound to M.
+                if (i == (NUMERICS.length - 1) && NUMERICS[i] <= n) {
+                    lowerBound = VALUES_TO_LETTERS.get(NUMERICS[i]);
+                } else if (NUMERICS[i] <= n && n < NUMERICS[i + 1]) {
+                    lowerBound = VALUES_TO_LETTERS.get(NUMERICS[i]);
+                    // break out of the loop as soon as lowerBound is determined
+                    break;
```

Comment:
You can eliminate this `break` by either:

1. adding `&& lowerBound != null` to the loop termination condition.
2. pulling the loop out into a separate procedure (from which you return when a match is found)